deltacat/catalog/v2/catalog_impl.py

```diff
@@ -0,0 +1,326 @@
+    namespace: Optional[str] = None,
+    mode: TableWriteMode = TableWriteMode.AUTO,
+    content_type: ContentType = ContentType.PARQUET,
+    *args,
+    **kwargs,
+) -> None:
+    raise NotImplementedError("Not implemented")
+
+
+def read_table(
+    table: str, namespace: Optional[str] = None, *args, **kwargs
+) -> DistributedDataset:
+    raise NotImplementedError("Not implemented")
+
+
+def alter_table(
+    table: str,
+    namespace: Optional[str] = None,
+    lifecycle_state: Optional[LifecycleState] = None,
+    schema_updates: Optional[Dict[str, Any]] = None,
+    partition_updates: Optional[Dict[str, Any]] = None,
+    sort_keys: Optional[SortScheme] = None,
+    description: Optional[str] = None,
+    properties: Optional[TableProperties] = None,
+    *args,
+    **kwargs,
+) -> None:
+    """Alter table definition."""
+    raise NotImplementedError("alter_table not implemented")
+
+
+def create_table(
+    table: str,
+    *args,
+    namespace: Optional[str] = None,
+    version: Optional[str] = None,
+    lifecycle_state: Optional[LifecycleState] = LifecycleState.ACTIVE,
+    schema: Optional[Schema] = None,
+    partition_scheme: Optional[PartitionScheme] = None,
+    sort_keys: Optional[SortScheme] = None,
+    description: Optional[str] = None,
+    table_properties: Optional[TableProperties] = None,
+    namespace_properties: Optional[NamespaceProperties] = None,
+    content_types: Optional[List[ContentType]] = None,
+    fail_if_exists: bool = True,
+    **kwargs,
+) -> TableDefinition:
+    """
+    Create an empty table. Raises an error if the table already exists and
+    `fail_if_exists` is True (default behavior).
+    """
+    catalog = kwargs.get("catalog")
+    if not isinstance(catalog, CatalogProperties):
+        raise ValueError("Catalog must be a CatalogProperties instance")
+
+    namespace = namespace or default_namespace()
+
+    # TODO
+    # Check if table exists
+    # if table_exists(table, namespace, **kwargs):
+    #     if fail_if_exists:
+    #         raise ValueError(f"Table {namespace}.{table} already exists")
+    #     return get_table(table, namespace, catalog=catalog)
+
+    # Create namespace if it doesn't exist
+    if not namespace_exists(namespace, catalog=catalog):
+        create_namespace(
+            namespace=namespace, properties=namespace_properties, catalog=catalog
+        )
+
+    # Extract merge keys from sort keys if provided
+    merge_keys = []
+    if sort_keys:
+        merge_keys = [key.name for key in sort_keys.keys]
+
+    # Create a table version through the storage layer
+    storage_impl.create_table_version(
+        namespace=namespace,
+        table_name=table,
+        schema=schema,
+        partition_scheme=partition_scheme,
+        sort_keys=sort_keys,
+        table_version_description=description,
+        table_description=description,
+        table_properties=table_properties,
+        lifecycle_state=lifecycle_state or LifecycleState.ACTIVE,
```

Comment:
I feel like a transition of a new table version to `ACTIVE` makes sense by default as part of an atomic data write transaction that also happened to create a table, but I'm not so sure when creating an empty table version. Does it make more sense for an empty table version be left in a `CREATED` lifecycle state by default and then be manually transitioned to `ACTIVE` to telegraph that it's ready for consumption?